### PR TITLE
Update Document as per the NGSI-LD v1.3.1

### PR DESCRIPTION
**Pull request by mohitNEC**

### CODE REVIEW

File: docs/en/source/API_walkthrough.rst [hunk 1]
```diff
@@ -146,7 +146,7 @@ First, we are going to create house2:smartrooms:room1. Let's assume that at enti
 		"type": "Relationship",
 		"object": "smartcity:houses:house2"
 	  },
-	  "@context": [{"Room": "urn:mytypes:room", "temperature": "myuniqueuri:temperature", "isPartOf": "myuniqueuri:isPartOf"},"https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"]
```
Comment (amitrnec) [inline docs/en/source/API_walkthrough.rst:149]: [Minor] why are you remove this @context. Please revert the changes same as previous.

File: docs/en/source/API_walkthrough.rst [hunk 2]
```diff
@@ -177,7 +177,7 @@ Next, let's create house2:smartrooms:room2 in a similar way.
...
```
Comment (amitrnec) [inline docs/en/source/API_walkthrough.rst:180]: [Minor] same as above

File: docs/en/source/API_walkthrough.rst [hunk 3]
```diff
@@ -236,14 +236,12 @@ If we want to just get the house in our example we would do a GET request like t
 	curl localhost:9090/ngsi-ld/v1/entities/smartcity%3Ahouses%3Ahouse2 -s -S -H 'Accept: application/ld+json' 
 
 Mind the url encoding here, i.e. ':' gets replaced by %3A. For consistency you should always encode your URLs. 
-
-Since we didn't provide our own @context in this request, only the parts of the core context will be replaced in the reply.
 ::
 
 	{
 		"id": "smartcity:houses:house2",
-		"type": "urn:mytypes:house",
```
Comment (amitrnec) [inline docs/en/source/API_walkthrough.rst]: [Minor] same as above.

File: docs/en/source/API_walkthrough.rst [hunk 4]
```diff
@@ -267,30 +265,13 @@ Since we didn't provide our own @context in this request, only the parts of the
 				"coordinates": [-8.50000005, 41.2]
 			}
 		}
-		"@context": ["https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"]
-	}
-
-As you can see entrance was compacted properly since it is was prefixed from the default context specified in the core context.
```
Comment (amitrnec) [inline docs/en/source/API_walkthrough.rst:273]: [Minor] please revert these changes same as previous.

File: docs/en/source/API_walkthrough.rst [hunk 5]
```diff
@@ -267,30 +265,13 @@ Since we didn't provide our own @context in this request, only the parts of the
...
-			"temperature": "myuniqueuri:temperature",
-			"isPartOf": "myuniqueuri:isPartOf"
-		}, "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"]
+		"@context": ["https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.3.jsonld"]
 	}
 
 We repeat this call providing our @context via the 'Link' like this 
 ::
 
-	curl localhost:9090/ngsi-ld/v1/entities/smartcity%3Ahouses%3Ahouse2 -s -S -H 'Accept: application/ld+json' -H 'Link: <https://pastebin.com/raw/Mgxv2ykn>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"'
```
Comment (amitrnec) [inline docs/en/source/API_walkthrough.rst:293]: [Trival] add pastebin url here same as previous.

File: docs/en/source/API_walkthrough.rst [hunk 6]
```diff
@@ -628,7 +607,7 @@ In order to append a new attribute to an entity you execute an HTTP PATCH comman
 Append in NGSI-LD by default will overwrite an existing entry. If this is not desired you can add the option parameter with noOverwrite to the URL like this /entities/<entityId>/attrs?options=noOverwrite. Now if we want to add an additional entry for the humidity in room1 we do an HTTP PATCH like this. 
 ::
 
-	curl localhost:9090/ngsi-ld/v1/entities/house2%3Asmartrooms%3Aroom1/attrs -s -S -X PATCH -H 'Content-Type: application/json' -H 'Link: https://pastebin.com/raw/Mgxv2ykn' -d @- <<EOF
+	curl localhost:9090/ngsi-ld/v1/entities/house2%3Asmartrooms%3Aroom1/attrs -s -S -X PATCH -H 'Content-Type: application/json' -H 'Link: https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.3.jsonld' -d @- <<EOF
 	{
 		"humidity": {
 		"value": 34,
```
Comment (amitrnec) [inline docs/en/source/API_walkthrough.rst:633]: [Trival] Add "EOF"

File: docs/en/source/API_walkthrough.rst (hunk 1 above)
Comment (mohitNEC) [reply docs/en/source/API_walkthrough.rst:149]: I have reverted the changes same as previous. Please review again

File: docs/en/source/API_walkthrough.rst (hunk 2 above)
Comment (mohitNEC) [reply docs/en/source/API_walkthrough.rst:180]: Done

File: docs/en/source/API_walkthrough.rst (hunk 3 above)
Comment (mohitNEC) [reply docs/en/source/API_walkthrough.rst]: Done

File: docs/en/source/API_walkthrough.rst (hunk 4 above)
Comment (mohitNEC) [reply docs/en/source/API_walkthrough.rst:273]: Done

File: docs/en/source/API_walkthrough.rst (hunk 5 above)
Comment (mohitNEC) [reply docs/en/source/API_walkthrough.rst:293]: Done

File: docs/en/source/API_walkthrough.rst (hunk 6 above)
Comment (mohitNEC) [reply docs/en/source/API_walkthrough.rst:633]: Done

File: docs/en/source/callFlow.rst [hunk 7]
```diff
@@ -13,41 +13,33 @@ The Figure is showing the operational flow of entity create/update/append in the
...
 
- 6.1. So it prepares the csource registration payload (as per NGSI_LD spec section C.3) from the entity payload and fills the necessary field (like id, endpoint as broker IP, location, etc.). Afterword entity manager writes this created csource payload in the CR Topic.
+6.1 Entity topic is published with the same payload so afterwards entity topic writes this created context source payload in the Registry Topic.
 
- 6.2.CR Manager listens to this CR topic and then able to know that some entity has registered.
+6.2 Context Source Registry listens to this registry topic and then able to know that some entity has registered.
 
- 6.3.CR manager writes the updates, if any are there, into the Csource Topic.  
+6.3 Context Source Registry writes the updates, if any are there, into the CSource Topic.
```
Comment (amitrnec) [inline docs/en/source/callFlow.rst]: [Major] Please changes in flow diagram 1
- Change arrow direction in flow no 6.2
- Add row direction subscription manager to I_REGSUB topic
- Remove CSource topic direction
-  Also changes in flow diagram theoretical part same as figure.

File: docs/en/source/callFlow.rst [hunk 8]
```diff
@@ -60,40 +52,41 @@ The Figure is showing the operational flow of entity subscription in the Scorpio
...
-Limitation: In the first release of Scorpio Broker, Csource query is not supported instead csource query is based on the internal messaging queue mechanism. In the future, both the message queue and Rest based csource query would be supported.
+ 6.2. Context Source Registry publishes the list of context sources into the CRQueryResult topic upon which the SM would have already started listening and repeat steps 5.2.3 and 5.2.4.
+
+**Note**: CSource Topic will contain the list of context sources registered through Csource registration interface directly. Registry Topic will contain the map of Entity Data model (maintained as an entity ID) created based on entity creation request (through IoT broker interface) and/or provider/data source of that entity model. Limitation: In the first release of Scorpio Broker, Csource query is not supported instead csource query is based on the internal messaging queue mechanism. In the future, both the message queue and Rest based csource query would be supported.
```
Comment (amitrnec) [inline docs/en/source/callFlow.rst]: [Major] Please changes in flow diagram 2
- Remove CR Query Result topic and CR Query topic direction
- Remove CSource topic direction
- Change Registry topic direction same as above
- Add direction from I_REGSUB topic to registration subscription manager
- Add direction from  I_NOTIFY topic to subscription manager
-  Add direction from ENTITY topic to subscription manager
-  Add changes flow diagram theoretical part.

File: docs/en/source/callFlow.rst [hunk 9]
```diff
@@ -102,33 +95,33 @@ Query
...
+6. Now, query response is being build after fetching the data from DB for the requested query and afterwards our query response gets published in the query topic which is being listed by Query Manager.
 
-7. The QM receives the notification from the QueryResult topic. 
+7. The QM receives the notification from the QueryResult topic.
 
- 7.1.  It sends the HTTP response back to the API gateway.
+ 7.1. It sends the HTTP response back to the API gateway.
 
- 7.2.API gateway sends back the response to the end-user/requestor.
+ 7.2. API gateway sends back the response to the end-user/requestor.
```
Comment (amitrnec) [inline docs/en/source/callFlow.rst]: [Minor] Add remove all the topic direction like as Query Result, CSource from flow diagram 3 and changes/remove from flow diagram theoretical part as well.

File: docs/en/source/callFlow.rst [hunk 10]
```diff
@@ -145,27 +138,29 @@ The Figure is showing the operational flow of context source registration in the
...
- 5.1 sends the Http response back to the API gateway.
+6. Now, Context Source registry prepares the response for csource registration request.
 
- 5.2 API gateway sends back the response to the end-user/requester.
+ 6.1 sends the Http response back to the API gateway.
 
-**Note**: For Conext Source Update request only the payload will get changes and in step 3 upon validation for the existing entity it will not exit rather it will update the retrieved entity and write it back into the Kafka. The rest of the flow will remain mostly the same. 
+ 6.2 API gateway sends back the response to the end-user/requester.
+
+**Note**: For Conext Source Update request only the payload will get changes and in step 3 upon validation for the existing entity it will not exit rather it will update the retrieved entity and write it back into the Kafka. The rest of the flow will remain mostly the same.
```
Comment (amitrnec) [inline docs/en/source/callFlow.rst]: [Major] Please changes in flow diagram 4
- Remove CSource and CSource sub topics direction 
-  Registry topic direction same as above
-   Add from Registry topic direction to Registry Subscription manager
-   Changes in theoretical part as well

File: docs/en/source/callFlow.rst [hunk 11]
```diff
@@ -175,33 +170,35 @@ Context Source Subscription
...
+5. Registry Subscription Manager will keep listening to the CSourceSub topic and for any new entry.
 
-5. The CR manager prepares the response for csource subscription request and
+6. The Registry Subscription manager prepares the response for csource subscription request and
 
- 5.1 sends the Http response back to the API gateway.
+ 6.1 sends the Http response back to the API gateway.
 
- 5.2 API gateway sends back the response to the end-user/requester.
+ 6.2 API gateway sends back the response to the end-user/requester.
```
Comment (amitrnec) [inline docs/en/source/callFlow.rst]: [Major] Please changes in flow diagram 5
- Remove topic direction same as above
-  Add direction from registry subscription manager to  I_NOTIFY topic
-  Add direction from  I_NOTIFY topic to subscription manager
-  Changes in theoretical part same as flow diagram direction.

File: docs/en/source/callFlow.rst [hunk 12]
```diff
@@ -211,18 +208,18 @@ History
...
+ 2.2 The service API gateway forwards the HTTP request to the History Manager micro-service.
 
-3. The history manager now executes the EVA algorithm approach on the received payload and push payload attributes to Kafka topic “TEMPORALENTITY”.
+3. The history manager now executes the EVA algorithm approach on the received payload and push payload attributes to Kafka topic “TEMPORAL”.
 
 **Note**: History Manager must walk through each attribute at the root level of the object (except @id and @type). Inside each attribute, it must walk through each instance (array element). Then, it sends the current object to the Kafka topic TEMPORALENTITY.
 
-4. The history manager will keep listening to the “TEMPORALENTITY” topic and for any new entry and performs the relative operation in the database.
\ No newline at end of file
+4. The history manager will keep listening to the “TEMPORAL” topic and for any new entry and performs the relative operation in the database.
```
Comment (amitrnec) [inline docs/en/source/callFlow.rst:202]: [Minor] Please changes in flow diagram 6 same as above.

File: docs/en/source/post_queries.rst [hunk 13]
```diff
@@ -0,0 +1,115 @@
+************
+POST QUERIES
+************
+
+The reason to provide a way to query NGSI_LD entities via POST, is using GET:
+
+1. We end up assembling very long URLs,due to URI parameters for 'id', 'attrs', etc, being included in the URL.
+2. By using POST, there is no need of long URLs.
+
+So, the difference lies in that instead of passing the inputs as URI parameters, we in POST Query service passes all the query items in the payload body.
```
Comment (martin-p-bauer) [inline docs/en/source/post_queries.rst]: ..., in a POST Query, we pass all the query items in the payload body.

File: docs/en/source/post_queries.rst [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+DESIGN
+######
+
+We created a novel **/entityOperations/query** endpoint in the API
```
Comment (martin-p-bauer) [inline docs/en/source/post_queries.rst]: "new" instead of "novel"

File: docs/en/source/post_queries.rst [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+As we can see this new POST query operation has just one single query as input.
+
+ - It is not allowed to send multiple queries in a single request as it is not consideres as a "BATCH Operation"
```
Comment (martin-p-bauer) [inline docs/en/source/post_queries.rst]: considereD

File: docs/en/source/post_queries.rst [hunk 16]
```diff
@@ -0,0 +1,115 @@
+************
+POST QUERIES
+************
+
+The reason to provide a way to query NGSI_LD entities via POST, is using GET:
```
Comment (martin-p-bauer) [inline docs/en/source/post_queries.rst]: .... is that using GET has the following issue:

File: docs/en/source/post_queries.rst [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+   * - temporalQ
+     - TemporalQuery
+     - A valid temporal operation
+     - 0.1
+	 
+	 
+POST Query Example
+##################
+
+Here is an example of Request URL and Payload data that how we can pass the query in the Payload.
```
Comment (martin-p-bauer) [inline docs/en/source/post_queries.rst]: ... data that SHOWS how we can pass ...

**Review by martin-p-bauer — CHANGES_REQUESTED** [review]
Please fix the identified language issues

File: docs/en/source/callFlow.rst (hunk 7 above)
Comment (mohitNEC) [reply docs/en/source/callFlow.rst]: Done

File: docs/en/source/callFlow.rst (hunk 8 above)
Comment (mohitNEC) [reply docs/en/source/callFlow.rst]: Done

File: docs/en/source/callFlow.rst (hunk 9 above)
Comment (mohitNEC) [reply docs/en/source/callFlow.rst]: Done

File: docs/en/source/callFlow.rst (hunk 10 above)
Comment (mohitNEC) [reply docs/en/source/callFlow.rst]: Done

File: docs/en/source/callFlow.rst (hunk 11 above)
Comment (mohitNEC) [reply docs/en/source/callFlow.rst]: Done

File: docs/en/source/callFlow.rst (hunk 12 above)
Comment (mohitNEC) [reply docs/en/source/callFlow.rst:202]: Done

File: docs/en/source/post_queries.rst (hunk 13 above)
Comment (mohitNEC) [reply docs/en/source/post_queries.rst]: Done

File: docs/en/source/post_queries.rst (hunk 14 above)
Comment (mohitNEC) [reply docs/en/source/post_queries.rst]: Done

File: docs/en/source/post_queries.rst (hunk 15 above)
Comment (mohitNEC) [reply docs/en/source/post_queries.rst]: Done

File: docs/en/source/post_queries.rst (hunk 16 above)
Comment (mohitNEC) [reply docs/en/source/post_queries.rst]: Done

File: docs/en/source/post_queries.rst (hunk 17 above)
Comment (mohitNEC) [reply docs/en/source/post_queries.rst]: Done
